File: flutter/vex_owl_admin/lib/core/services/auth_service.dart
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:shared_preferences/shared_preferences.dart';
import '../../core/constants/app_constants.dart';
import '../../core/constants/api_constants.dart';
import '../../core/utils/api_client.dart';
import '../../core/utils/storage_utils.dart';
import '../../data/models/login_models.dart';

final authServiceProvider = Provider<AuthService>((ref) {
  return AuthService(ref.read(apiClientProvider));
});

class AuthService {
  final ApiClient _apiClient;

  AuthService(this._apiClient);

  Future<TokenResponse> login(LoginRequest request) async {
    final response = await _apiClient.post<TokenResponse>(
      ApiConstants.authLogin,
      data: request.toJson(),
      parser: (data) => TokenResponse.fromJson(data),
    );
    final token = response.token;
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(AppConstants.tokenKey, token);
    await prefs.setString(AppConstants.userKey, response.subjectId);
    return response;
  }

  Future<void> logout() async {
    await StorageUtils.clearAuth();
  }

  Future<bool> isLoggedIn() async {
    final prefs = await SharedPreferences.getInstance();
    final token = prefs.getString(AppConstants.tokenKey);
    return token != null && token.isNotEmpty;
  }
}

File: owl-admin/src/core/services/authService.ts
import { apiClient, ApiException } from '../utils/apiClient';
import { storage } from '../utils/storage';
import { ApiConstants } from '../constants/apiConstants';
import { LoginRequest, TokenResponse } from '../../data/models/loginModels';

export class AuthService {
  async login(request: LoginRequest): Promise<TokenResponse> {
    try {
      const response = await apiClient.post<TokenResponse>(ApiConstants.authLogin, {
        principal: request.principal,
        credentials: request.credentials,
        loginType: request.loginType || 'admin',
      });
      storage.setToken(response.token);
      storage.setUserId(response.subjectId);
      return response;
    } catch (error) {
      if (error instanceof ApiException) {
        throw error;
      }
      throw new ApiException('登录失败，请稍后重试');
    }
  }

  async logout(): Promise<void> {
    storage.clearAuth();
  }

  isLoggedIn(): boolean {
    return storage.isLoggedIn();
  }
}

export const authService = new AuthService();